javaparser-core/src/main/java/com/github/javaparser/JavaParser.java

```diff
@@ -128,7 +132,7 @@ private GeneratedJavaParser getParserForProvider(Provider provider) {
         } finally {
             try {
                 provider.close();
-            } catch (IOException e) {
+            } catch (@SuppressWarnings("squid:S1166") IOException e) {
                 // Since we're done parsing and have our result, we don't care about any errors.
```

Comment:
This project does not use Sonar, so no need to suppress its warnings. Besides that: there is a clear explanation on the next line.